mkosi/burn.py

```diff
@@ -0,0 +1,44 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+import os
+import sys
+
+from mkosi.config import (
+    MkosiArgs,
+    MkosiConfig,
+    OutputFormat,
+)
+from mkosi.log import complete_step, die
+from mkosi.run import run
+
+
+def run_burn(args: MkosiArgs, config: MkosiConfig) -> None:
+    if config.output_format not in (OutputFormat.disk, OutputFormat.esp):
+        die(f"{config.output_format} images cannot be dd'ed to disk")
```

Comment:
Since dd isn't used anymore, maybe
```suggestion
        die(f"{config.output_format} images cannot be burned to disk")
```